src/core/execute.c

```diff
@@ -2369,7 +2374,7 @@ static int apply_mount_namespace(
                             &ns_info, context->read_write_paths,
                             needs_sandboxing ? context->read_only_paths : NULL,
                             needs_sandboxing ? context->inaccessible_paths : NULL,
-                            empty_directories,
+                            empty_directories ? : context->empty_directories,
```

Comment:
Oh, is this why compile_bind_mounts() doesn't return empty_directories when context->empty_directories is the only one?

That's kind of ugly IMHO.

Reply:
It might be. But this logic can reduce creating unnecessary copy of `context->empty_directories`.